code/modules/assembly/health.dm

```diff
@@ -0,0 +1,105 @@
+/obj/item/device/assembly/health
+	name = "health sensor"
+	desc = "Used for scanning and monitoring health."
+	icon_state = "health"
+	materials = list(MAT_METAL=800, MAT_GLASS=200)
+	origin_tech = "magnets=1;biotech=1"
+	attachable = 1
+	secured = 0
+
+	var/scanning = 0
+	var/health_scan
+	var/alarm_health = 0
+
+
+
+/obj/item/device/assembly/health/activate()
+	if(!..())	return 0//Cooldown check
+	toggle_scan()
+	return 0
+
+/obj/item/device/assembly/health/toggle_secure()
+	secured = !secured
+	if(secured && scanning)
+		SSobj.processing |= src
+	else
+		scanning = 0
+		SSobj.processing.Remove(src)
+	update_icon()
+	return secured
+
+/obj/item/device/assembly/health/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/device/multitool))
+		if(alarm_health == 0)
+			alarm_health = -90
+			user.show_message("You toggle [src] to \"detect death\" mode.")
+		else
+			alarm_health = 0
+			user.show_message("You toggle [src] to \"detect critical state\" mode.")
+		return
+	else
+		return ..()
+
+/obj/item/device/assembly/health/process()
+	if(!scanning || !secured)
+		return
+
+	var/atom/A = src
+	if(connected && connected.holder)
+		A = connected.holder
+
+	for(A, A && !ismob(A), A=A.loc);
```

Comment:
Ah yes, true enough, I wasn't reading, 12 hours ago would put it at the exact hour I woke up, probably within 10 minutes haha.